### PR TITLE
Fix `cart_quantity` in front ProductController

### PR DESCRIPTION
**Pull request by boherm**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Fix `cart_quantity`, we don't have `$product['id_product']` anymore in the array, so we can't calculate anymore the `cart_quantity` in product detail page.
| Type?             | bug fix
| Category?         | FO
| BC breaks?        | no
| Deprecations?     | no
| How to test?      | 1. Set a minimal quantity in a combination product (for one combination only!)<br>2. Go to the shop, and add in cart the combination product setting with a new minimum quantity.<br>3. When we close the confirmation modal, we need to see that the input quantity is set to 1 instead of the minimal quantity
| UI Tests          | https://github.com/boherm/ga.tests.ui.pr/actions/runs/9746450163
| Fixed issue or discussion?     | Related to observation made by @tblivet in https://github.com/PrestaShop/PrestaShop/issues/36456
| Related PRs       | https://github.com/PrestaShop/PrestaShop/pull/35902
| Sponsor company   | PrestaShop SA

### CODE REVIEW

**Review by nicosomb — APPROVED** [review]

**Review by matthieu-rolland — APPROVED** [review]

**Review by florine2623 — APPROVED** [review]
Hello @boherm ,

Tested on all types of products.
Tested standard case with combinations.
Tested with classic and hummingbird.

https://github.com/PrestaShop/PrestaShop/assets/16019289/17aa193e-cad6-4953-8508-7d944d4e59d2




LGTM ✅ 
It is QA ✅

**Comment by ps-jarvis** [comment]
PR merged, well done!

Message to @PrestaShop/committers: do not forget to milestone it before the merge.

**Pull request merged**
